lib/broadway/topology/rate_limiter.ex

```diff
@@ -16,20 +14,20 @@ defmodule Broadway.Topology.RateLimiter do
         name = Keyword.fetch!(opts, :name)
         producers_names = Keyword.fetch!(opts, :producers_names)
         args = {name, rate_limiting_opts, producers_names}
-        GenServer.start_link(__MODULE__, args, name: table_name(name))
+        GenServer.start_link(__MODULE__, args, name: rate_limiter_name(name))
     end
   end
 
-  def rate_limit(table_name, amount)
-      when is_atom(table_name) and is_integer(amount) and amount > 0 do
-    :ets.update_counter(table_name, @row_name, -amount)
+  def rate_limit(counter, amount)
+      when is_reference(counter) and is_integer(amount) and amount > 0 do
+    :atomics.sub_get(counter, 1, amount)
   end
 
-  def get_currently_allowed(table_name) when is_atom(table_name) do
-    :ets.lookup_element(table_name, @row_name, 2)
+  def get_currently_allowed(counter) when is_reference(counter) do
+    :atomics.get(counter, 1)
```

Comment:
With my suggestion above, this would become
```suggestion
    :atomics.get(counter, @atomics_index)
```